backend/src/main/java/com/redhat/cloud/notifications/routers/internal/errata/ErrataMigrationRepository.java

```diff
@@ -0,0 +1,104 @@
+package com.redhat.cloud.notifications.routers.internal.errata;
+
+import com.redhat.cloud.notifications.models.EventType;
+import com.redhat.cloud.notifications.models.EventTypeEmailSubscription;
+import com.redhat.cloud.notifications.models.EventTypeEmailSubscriptionId;
+import com.redhat.cloud.notifications.models.SubscriptionType;
+import io.quarkus.logging.Log;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import jakarta.persistence.EntityManager;
+import jakarta.transaction.Transactional;
+import jakarta.validation.ConstraintViolationException;
+
+import java.util.List;
+
+@ApplicationScoped
+public class ErrataMigrationRepository {
+    /**
+     * The maximum size of the batch of subscriptions we want to insert in the
+     * database.
+     */
+    private static final int INSERTION_BATCH_SIZE = 500;
+
+    /**
+     * The internal name of the Errata application in our database.
+     */
+    public static final String ERRATA_APPLICATION_NAME = "errata-notifications";
+
+    @Inject
+    EntityManager entityManager;
```

Comment:
We could use a `StatelessSession` in this class:
```java
@Inject
StatelessSession statelessSession;
```
https://thorben-janssen.com/hibernates-statelesssession/

Reply:
I have tried that but when executing the code the `this.statelessSession.begin()` statement gives me the following error:

```
java.lang.IllegalStateException: Not supported for JTA entity managers
```

I have been trying to find information about this but with no luck. Does it complain because we are using `ResourceHelpers` that are creating stuff via the `EntityManager`? Do I have to manually create stuff for this instead? :thinking:

Reply:
Done!